olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -218,11 +237,10 @@ boost::any DefaultCacheImpl::Get(const std::string& key,
     return boost::any();
   }
 
-  PromoteKeyLru(key);
-
   if (memory_cache_) {
     auto value = memory_cache_->Get(key);
     if (!value.empty()) {
+      PromoteKeyLru(key);
```

Comment:
Why under memory_cache? What if the value is not in the memory cache and needs to be taken from the mutable_cache_? Then you do not promote that key in LRU. This seems not right to me.

Reply:
Its promoted in that case too(in `GetFromDiscCache()` method).